### ISSUE TITLE
Prevent unintentional element movement at selection

### ISSUE THREAD

**Issue by SrGnis** [excalidraw/excalidraw#8713]
I have been using exalidraw the last few days and there is one little thing that bothers me, if you click on an element and your mouse is not completely still the element will end up moving a little bit with the mouse.

Theoretically this is working correctly but the amount of times I have double clicked on an element and in the process the mouse has moved a pixel and therefore the element has also moved with it is quite high.

A possible solution would be to add a small timeout from the click on the element until it starts processing the mouse movement. 
Maybe 0.3 seconds or so.

**Comment by pahulgogna** [comment]
I believe that it would be better to add a small buffer to moving an element from the initial position, so that you can also snap the element back to its initial position instead of adding a timeout every time you select an element.

I would like to start working on this feature.

**Comment by Mrazator** [comment]
> I believe that it would be better to add a small buffer to moving an element from the initial position

Yeah, likely `setTimeout` isn't the best here. Btw. do you mean detecting a sudden slight movement (delta) and ignoring that? IIRC we are already doing something similar for various tiny pen/trackpad movements.

> I would like to start working on this feature.

Sure, it's yours, but keep in mind some slight movements might be intentional:
1. the "small buffer" needs to be relative to the zoom level for all levels - i.e.  when zoomed in a lot, even a small delta might be intentional
2. when snapping, prefer the snapping tool
3. when grid is on, prefer grid alignment
4. when using keyboard arrows

**Comment by pahulgogna** [comment]
>Btw. do you mean detecting a sudden slight movement (delta) and ignoring that?

Yes, but only initially, just like it is in this video editing software in the video below,

https://github.com/user-attachments/assets/c09e4f7a-64c9-4e04-a543-11424faf2914

>Sure, it's yours, but keep in mind some slight movements might be intentional:

Alright! These points were really helpful, there were some things I had not considered. Thank You.

**Comment by dwelle** [comment]
> Yes, but only initially, just like it is in this video editing software in the video below,

Way too big a threshold.

The small, albeit significant friction this adds was the reason why we haven't implemented it yet, because all these small things together change how Excalidraw feels.

I think for us to do it, we'd have to come up with some better heuristic than just "don't drag until you move X pixels".

Some ideas to make it at least a bit smarter:

1. if element is already selected before you `pointerdown`, don't enable any drag thresholds
2. bigger drag-threshold (if not enable solely) if you don't move your cursor much _before_ you `pointerdown` which would indicate selection intent as opposed to drag intent. In other words, if user moves cursor a significant distance right before they click, it could suggest their intention is to further keep moving the cursor after they select the element.

**Comment by pahulgogna** [comment]
Hey, I've just opened a PR for this issue.

Could you provide feedback on whether I should continue with this approach and incorporate the suggestions from @Mrazator, or would it be better to pivot and explore tracking mouse movement in real-time to determine how the element behaves on selection instead?

**Comment by SrGnis** [comment]
Hello, I need to do something here? I tried the demo and works fine.

**Comment by KappyDays** [comment]
I’m still experiencing this issue with the current version when selecting objects with a mouse. Even a very small movement between pointer-down and pointer-up can unintentionally reposition an element, especially when double-clicking text.

Is there currently any workaround or preference that reduces this sensitivity? I also noticed that #8719 is still open—has there been any updated direction on this issue? An optional per-user drag threshold would be very helpful even if the default behavior remains unchanged.

Thanks!
